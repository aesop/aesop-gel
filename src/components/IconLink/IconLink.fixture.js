/* istanbul ignore file */
const IconLinkFixture = {
  items: [
    {
      altText: 'Instagram',
      icon: {
        altText: 'instagram',
        large: './assets/images/IconLink/instagram.png',
        medium: './assets/images/IconLink/instagram.png',
        small: './assets/images/IconLink/instagram.png',
      },
      target: 'https://about.instagram.com/',
    },
    {
      altText: 'Twitter',
      icon: {
        altText: 'Wwitter',
        large: './assets/images/IconLink/twitter.png',
        medium: './assets/images/IconLink/twitter.png',
        small: './assets/images/IconLink/twitter.png',
      },
      target: 'https://twitter.com/home?lang=en',
    },
    {
      altText: 'LinkedIn',
      icon: {
        altText: 'linkedin',
        large: './assets/images/IconLink/linkedin.png',
        medium: './assets/images/IconLink/linkedin.png',
        small: './assets/images/IconLink/linkedin.png',
      },
      target: 'https://www.linkedin.com/home/?originalSubdomain=au',
    },
    {
      altText: 'Kakao',
      icon: {
        altText: 'kakao',
        large: './assets/images/IconLink/kakao.png',
        medium: './assets/images/IconLink/kakao.png',
        small: './assets/images/IconLink/kakao.png',
      },
      target: 'https://www.kakaocorp.com/service/KakaoTalk?lang=en',
    },
  ],
};

export { IconLinkFixture };
