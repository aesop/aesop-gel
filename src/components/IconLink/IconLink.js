import React from 'react';
import PropTypes from 'prop-types';
import Hyperlink from '~/components/Hyperlink';
import Image from '~/components/Image';

const IconLink = ({
  altText,
  className,
  hasTargetInNewWindow,
  height,
  icon,
  iconImageClassName,
  id,
  target,
  width,
}) => {
  return (
    <Hyperlink
      className={className}
      dataTestRef={`DATA_TEST_REF_ICON_LINK_${id}`}
      hasTargetInNewWindow={hasTargetInNewWindow}
      title={altText}
      url={target}
    >
      <Image
        altText={altText}
        className={iconImageClassName}
        large={icon.large}
        medium={icon.medium}
        small={icon.small}
        style={{ height: `${height}px`, width: `${width}px` }}
      />
    </Hyperlink>
  );
};

IconLink.propTypes = {
  altText: PropTypes.string,
  className: PropTypes.string,
  hasTargetInNewWindow: PropTypes.bool,
  height: PropTypes.number,
  icon: PropTypes.object,
  iconImageClassName: PropTypes.string,
  id: PropTypes.string,
  target: PropTypes.string,
  width: PropTypes.number,
};

IconLink.defaultProps = {
  altText: undefined,
  className: undefined,
  hasTargetInNewWindow: true,
  height: 21,
  icon: undefined,
  iconImageClassName: undefined,
  id: undefined,
  target: undefined,
  width: 21,
};

export default IconLink;
