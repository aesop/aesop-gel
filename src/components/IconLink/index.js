/* istanbul ignore file */
import { IconLink } from './IconLink';

export { IconLink };
