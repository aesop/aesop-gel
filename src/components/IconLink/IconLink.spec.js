import React from 'react';
import { render } from '@testing-library/react';
import IconLink from './IconLink';
import IconLinkFixture from './IconLink.fixture';

describe('<IconLink />', () => {
  it('should be defined', () => {
    expect(IconLink).toBeDefined();
  });

  it('should render base component correctly', () => {
    const { container } = render(
      <IconLink
        altText={IconLinkFixture[0].altText}
        dataTestRef="test-data-ref"
        hasTargetInNewWindow={true}
        height={25}
        icon={IconLinkFixture[0].icon}
        id={IconLinkFixture[0].altText}
        target={IconLinkFixture[0].target}
        width={25}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
