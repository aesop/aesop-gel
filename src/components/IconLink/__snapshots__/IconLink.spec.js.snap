// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconLink /> renders base component correctly 1`] = `
<a
  className="base left dark"
  data-test-ref="test-data-ref"
  href="https://about.instagram.com/"
  target="_blank"
  title="Instagram"
>
  <picture
    className="base picture dark"
    style={
      Object {
        "height": "25px",
        "width": "25px",
      }
    }
  >
    <source
      media="(min-width: 1025px)"
      srcSet="./assets/images/IconLink/instagram.png"
    />
    <source
      media="(min-width: 640px)"
      srcSet="./assets/images/IconLink/instagram.png"
    />
    <source
      media="(min-width: 0px)"
      srcSet="./assets/images/IconLink/instagram.png"
    />
    <img
      alt="Instagram"
    />
  </picture>
</a>
`;
